### PR TITLE
Feat/add options to store collections

### PR DESCRIPTION
**Pull request by sebastiannicolajsen**
### What
- Add options such as search param q, ordering options and update endpoint description accordingly.

### How
- Modify the `product-collection` repository and the remaining stack of services to accommodate these changes.

### Testing
- Update endpoint unit tests to test that:  uses q if given, uses limit and offset if given, uses order if given.
- Update integration tests for collections with the following cases: espects search using q, can order, can order and use q at the same time.

### CODE REVIEW

File: packages/medusa/src/api/routes/store/collections/list-collections.js [hunk 1]
```diff
@@ -1,37 +1,108 @@
...
+  const order = value.order
+  listConfig.order = {
+    [order.replace("-", "")]: order[0] === "-" ? "DESC" : "ASC",
   }
 
-  const [collections, count] = await productCollectionService.listAndCount(
-    selector,
-    listConfig
-  )
+  const collections = await collectionService.list(selector, listConfig)
```
Comment (srindom) [inline packages/medusa/src/api/routes/store/collections/list-collections.js:100]: Should be `listAndCount` so we get the actual global count based on the query

**Review by srindom — CHANGES_REQUESTED** [review]
Small fixes - can we also please add `(created_at|updated_at)[gt|lt|gte...]`

**Review by srindom — CHANGES_REQUESTED** [review]
Needs to be refactored to adhere to new TS API layer

File: packages/medusa/src/api/routes/store/collections/list-collections.js (hunk 1 above)
Comment (vuuzle) [reply packages/medusa/src/api/routes/store/collections/list-collections.js]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

**Pull request closed without merging**